software/apps/synth/main.c

```diff
@@ -13,27 +14,89 @@
 #include "runtime.h"
 #include "synchronization.h"
 
-#define N 16
+// Macros
+#define UNROLL (4)
+#define N (NUM_CORES * UNROLL)
```

Comment:
Is `N` really `(NUM_CORES * UNROLL)` or should it be `(NUM_CORES * BANKING_FACTOR)`?